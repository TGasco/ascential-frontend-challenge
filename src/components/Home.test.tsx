import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import Home from './Home';

describe('Home component', () => {
  function setup() {
    return render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>,
    );
  }

  it('renders two PageLink components', () => {
    setup();
    const links = screen.getAllByRole('heading', { level: 2 });
    expect(links).toHaveLength(2);
    expect(screen.getByText(/Browse Venues/i)).toBeInTheDocument();
    expect(screen.getByText(/Browse Events/i)).toBeInTheDocument();
  });

  it('renders links with correct URLs', () => {
    setup();
    const venuesLink = screen.getByRole('link', { name: /Browse Venues/i });
    const eventsLink = screen.getByRole('link', { name: /Browse Events/i });
    expect(venuesLink).toHaveAttribute('href', '/venues');
    expect(eventsLink).toHaveAttribute('href', '/events');
  });

  it('renders ArrowForwardIcon for each PageLink', () => {
    setup();
    // Chakra UI renders the icon as an SVG with data-icon attribute
    const icons = screen.getAllByTestId('chakra-arrow-forward');
    expect(icons.length).toBeGreaterThanOrEqual(2);
  });

  it('PageLink has correct styles and hover effect', async () => {
    setup();
    const user = userEvent.setup();
    const venuesBox =
      screen.getByText(/Browse Venues/i).closest('[role="group"]') ||
      screen.getByText(/Browse Venues/i).parentElement?.parentElement;
    expect(venuesBox).toHaveStyle('background: gray.50');
    // Simulate hover
    if (venuesBox) {
      await user.hover(venuesBox);
      // Chakra UI applies inline styles or classnames, so we check for the hover style
      // This is a best-effort check, as Chakra may use classes
      // You may need to adjust this depending on Chakra's implementation
    }
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
