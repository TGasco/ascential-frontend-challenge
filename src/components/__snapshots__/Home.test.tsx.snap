// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1mhuh46"
  >
    <div
      class="chakra-linkbox css-uhbnde"
    >
      <div
        class="css-1eqzliz"
      >
        <h2
          class="chakra-heading css-0"
        >
          <a
            class="chakra-linkbox__overlay css-1hnz6hu"
            href="/venues"
          >
            Browse Venues
          </a>
        </h2>
        <svg
          class="chakra-icon css-dvq678"
          data-testid="chakra-arrow-forward"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div
      class="chakra-linkbox css-uhbnde"
    >
      <div
        class="css-1eqzliz"
      >
        <h2
          class="chakra-heading css-0"
        >
          <a
            class="chakra-linkbox__overlay css-1hnz6hu"
            href="/events"
          >
            Browse Events
          </a>
        </h2>
        <svg
          class="chakra-icon css-dvq678"
          data-testid="chakra-arrow-forward"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
